specifications/cdsc-wg1-01.md

```diff
@@ -1 +1,292 @@
+
+<a id="referenced-technologies" href="#referenced-technologies" class="permalink">🔗</a> Referenced Technologies: "[HTTPS](https://www.rfc-editor.org/rfc/rfc9110#name-https-uri-scheme)", "[URL](https://www.rfc-editor.org/rfc/rfc3986.html#section-1.1.3)", "[Request Methods](https://www.rfc-editor.org/rfc/rfc9110#name-methods)", "[Status Codes](https://www.rfc-editor.org/rfc/rfc9110#name-status-codes)", "[JSON](https://www.rfc-editor.org/rfc/rfc8259)", "[Well-Known URI](https://www.rfc-editor.org/rfc/rfc5785)", "[ISO8601](https://www.iso.org/iso-8601-date-and-time-format.html)", "[OAuth](https://oauth.net/specs/)", "[GeoJSON](https://datatracker.ietf.org/doc/html/rfc7946)", <span style="background-color:yellow">TODO: add more as needed</span> are defined by their referenced standards documents.
+
+<a id="key-words" href="#key-words" class="permalink">🔗</a> Key Words: "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" are defined in accordance with [BCP 14](https://www.rfc-editor.org/info/bcp14).
+
+## 3. Metadata Endpoint <a id="metadata-endpoint" href="#metadata-endpoint" class="permalink">🔗</a>
+
+Servers supporting this specification SHALL offer one or more HTTPS URLs from which a Client can perform a GET request to retrieve Server metadata, and the Server will respond with the Server's metadata JSON object.
+
+For successful responses, Servers MUST respond with a "200 OK" Status Code and body containing a JSON object that is formatted as defined in [Metadata Object Format](#metadata-object-format). Cache-optimized responses, such as "304 Not Modified", to Client requests with cached value headers, such as `If-None-Match`, is also acceptable.
+
+For invalid requests and error responses, Servers MUST respond with the appropriate Response Code for the type of issue or error (e.g. "503 Service Unavailable" if the Server is temporarily down for maintenance).
+
+### 3.1. Metadata Well-Known URI <a id="metadata-well-known-uri" href="#metadata-well-known-uri" class="permalink">🔗</a>
+
+If the Server wishes to provide metadata publicly, the Server MUST support a metadata endpoint at the Well-Known URI `"/.well-known/carbon-data-spec.json"` on their hostname (e.g. https://example.com/.well-known/carbon-data-spec.json).
+
+Servers MAY additionally support other public and restricted URLs for special case or private metadata endpoints, as needed.
+
+### 3.2. Metadata Object Format <a id="metadata-object-format" href="#metadata-object-format" class="permalink">🔗</a>
+
+Metadata objects are formatted as JSON objects and contain named values, some of which are required and some of which are optional. The following values are included in the default list available in metadata objects.
```

Comment:
From @jhughes22 ([on May 25, 2023](https://github.com/carbon-data-specification/Customer-Data/pull/23#discussion_r1205759610)):

---

Would a unique identifier of some kind for the Server entity make any sense? I see the form EIA-860 identifier used for this occasionally for US utilities in other contexts. In this case the Server does not have to be a utility, so it would probably need to be an optional field. Not a strong suggestion here, just something to think about. 

---

Reply:
Hmmm, I think this will be tricky since Servers are self-published, so there's not really any centralized naming authority to prevent collisions. The EIA-860 seems to be limited to only specific entity types, so maybe we could have an extension that allows for including that value as a capability?